tests/rbd/test_rbd_groups_image_clone.py

```diff
@@ -0,0 +1,245 @@
+                log.info(
+                    f"Clone the image using group snapshot group: {group} snap: {group_snap}"
+                )
+                clone = image + "_clone"
+                clone_create_kw = {
+                    "source-snap-spec": pool + "/" + image,
+                    "dest-image-spec": pool + "/" + clone,
+                    "snap-id": snap_id,
+                    "rbd-default-clone-format": "2",
+                }
+                rbd1.clone(**clone_create_kw)
+
+                # Verify the cloned images exist in the mentioned pool along with it’s parent image
+                log.info(f"Validate clone image {clone} exist")
+                info_spec = {"image-or-snap-spec": f"{pool}/{clone}", "format": "json"}
+                out, err = rbd1.info(**info_spec)
+                if err:
+                    log.error(f"Error while fetching info for image {clone}")
+                    return 1
+                out_json = json.loads(out)
+                log.info(f"Image info: {out_json}")
+
+                # 12. Map the cloned images as a new block disk
+                log.info(f"Map the cloned image {clone} and run IO")
+                image_spec = f"{pool}/{clone}"
+                size = kw.get("size")
+                file_path = f"/tmp/{clone}_dir/{clone}_file"
+                mount_config = {
+                    "rbd": rbd,
+                    "client": client,
+                    "size": size,
+                    "file_path": file_path,
+                    "image_spec": image_spec,
+                    "io": True,
+                }
+                if map_and_mount_image(**mount_config):
+                    log.error(f"Map and mount failed for {image_spec}")
+                    return 1
+
+    return 0
+
+
+def run(**kw):
+    """Tests the group creation followed by cloning image from group snapshot.
+
+    Args:
+        **kw: test data
+    """
+    log.info("Running test CEPH-83594298 - Cloning an image from a Group Snapshot")
+
+    try:
+        if kw.get("client_node"):
+            client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+        else:
+            client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+        kw["do_not_create_image"] = True
+        rbd_obj = initial_rbd_config(**kw)
+        pool_types = rbd_obj.get("pool_types")
+        ret_val = test_rbd_groups_image_clone(rbd_obj=rbd_obj, client=client, **kw)
+        if ret_val == 0:
+            log.info("Testing RBD cloning image from group snapshot Passed")
+    except Exception as e:
+        log.error(
+            f"Testing RBD cloning image from group snapshot failed with Error: {str(e)}"
+        )
+        ret_val = 1
+    finally:
+        cluster_name = kw.get("ceph_cluster", {}).name
+        if "rbd_obj" not in locals():
+            rbd_obj = Rbd(client)
+        obj = {cluster_name: rbd_obj}
+        cleanup(pool_types=pool_types, multi_cluster_obj=obj, **kw)
```

Comment:
Here if pool_types is not defined (e.g., due to a failure in initial_rbd_config()), the cleanup call will throw an UnboundLocalError.
Suggestion: Initialize pool_types to an empty list or include a check before cleanup:

Reply:
Done